feature/gribi/otg_tests/mpls_in_udp/README.md

```diff
@@ -0,0 +1,378 @@
+              rules:
+                rule: 1
+                  config:
+                    sequence-id: 1
+                  ipv6:
+                    config:
+                      destination-address: "inner_ipv6_default"
+                  action:
+                    encapsulate-mpls-in-gre:              # TODO: add to OC model/PR in progress
+                      targets:
+                        target: "default_dst_1"
+                          config:
+                            id: "default_dst_1"
+                            network-instance: "DEFAULT"
+                            source-ip: "outer_ipv6_src"
+                            destination-ip: "outer_ipv6_dst_def"
+                            ip-ttl: outer_ip-ttl
+                            dscp: outer_dscp
+                            inner-ttl-min: 2
+```
+
+* Generate the policy forwarding configuration
+* Push the configuration to DUT using gnmi.Set with REPLACE option
+* Configure ATE port 1 with traffic flow which does not match any AFT next hop route
+* Generate traffic from ATE port 1 to ATE port 2
+* Validate ATE port 2 receives GRE traffic with correct inner and outer IPs
+
+### TE-18.1.3 - Decapsulation set by gRIBI
+
+This gRIBI content is used to perform MPLS in UDP decapsulation.
+
+```proto
+network_instances {
+  network_instance {
+    afts {
+      ipv6_unicast {
+        ipv6_entry {
+          prefix: "outer_loopback_ipv6"   # IPv6 match rule for the device loopback expected to receive MPLS in UDP packets
+          next-hop-group: 999
+        }
+      }
+      ipv4_unicast {
+        ipv4_entry {
+          prefix: "outer_loopback_ipv4"
+          next-hop-group: 999
+        }
+      }
+      next_hop_groups {
+        next_hop_group {
+          next_hop_group_id: "Decap"  # New OC path /network-instances/network-instance/afts/next-hop-groups/next-hop-group/state/next-hop-group-id
+          id: 999
+          next_hops {
+            next_hop: {
+              index: 990
+            }
+          }
+        }
+      }
+      next_hops {
+        next_hop {
+          index: 990
+          decapsulate_header: OPENCONFIG_AFT_TYPES:MPLS_IN_UDPV6      # TODO: Add OC path for this
+          # The device should decapsulate the UDP packet and use the MPLS header
+          # to determine the destination network-instance, then forward the packet
+          # based on the inner IP packet, matching an appropriate AFT entry.
+        }
+      }
+    }
+  }
+}
+```
+
+* Push the gRIBIB the policy forwarding configuration
+* Push the configuration to DUT using gnmi.Set with REPLACE option
+* Configure ATE port 1 with traffic flow which does not match any AFT next hop route
+* Generate traffic from ATE port 1 to ATE port 2
+* Validate ATE port 2 receives GRE traffic with correct inner and outer IPs
+
+### TE-18.1.5 Rewrite inner packet TTL=2 if inner TTL=1
+
+* Perform mpls-in-udp encapsulation, but add a condition that the DUT must
```

Comment:
or set to 1. In short if the incoming packet TTL is 1, the same value 1 should go out. setting 2 will set the ttl to 2.

Reply:
Darren Loher,
Could you please remove TTL modifications from the pull request.
We can continue discussion at #3409
https://github.com/openconfig/featureprofiles/pull/3409

Reply:
Done